src/opds_dumper.cpp

```diff
@@ -124,13 +124,51 @@ BooksData getBooksData(const Library* library, const std::vector<std::string>& b
   return booksData;
 }
 
+const std::map<std::string, std::string> iso639_3 = {
+    {"atj","atikamekw"},
+    {"azb","آذربایجان دیلی"},
+    {"bcl","central bikol"},
+    {"bgs","tagabawa"},
+    {"bxr","буряад хэлэн"},
+    {"cbk","chavacano"},
+    {"cdo","閩東語"},
+    {"diq","dimli"},
+    {"dty","डोटेली"},
+    {"eml","emiliân-rumagnōl"},
+    {"fbs","српскохрватски"},
+    {"kbp","kabɩyɛ"},
+    {"kld","Gamilaraay"},
+    {"lbe","лакку маз"},
+    {"lbj","ལ་དྭགས་སྐད་"},
+    {"map","Austronesian"},
+    {"mhr","марий йылме"},
+    {"mnw","ဘာသာမန်"},
+    {"myn","mayan"},
+    {"nah","nahuatl"},
+    {"nai","north American Indian"},
+    {"nds","plattdütsch"},
+    {"nrm","bhasa narom"},
+    {"olo","livvi"},
+    {"pih","Pitcairn-Norfolk"},
+    {"pnb","Western Panjabi"},
+    {"rmr","Caló"},
+    {"rmy","romani shib"},
+    {"roa","romance languages"},
+    {"twi","twi"},
+    {"ido","ido"}
+};
+
 std::string getLanguageSelfName(const std::string& lang) {
   const icu::Locale locale(lang.c_str());
   icu::UnicodeString ustring;
   locale.getDisplayLanguage(locale, ustring);
   std::string result;
   ustring.toUTF8String(result);
-  return result;
+  try {
+    return iso639_3.at(lang);
+  } catch (...) {
+    return result;
```

Comment:
A small comment (I already made on Slack) on my side is that if we could call `iso639_3.at(lang)` only if necessary, this would be better as well. This is a bit more code, but would make it slighly more performant, and this routine is really called often.

Reply:
Such decisions should be substantiated by benchmarking data. But my bet is that it shouldn't matter much since `getLanguageSelfName()` is hardly going to be a bottleneck at all. Benchmarking can help to eliminate any concern with respect to the performance of  `getLanguageSelfName()` and just use the simplest solution (which you have already implemented).